get_git.go

```diff
@@ -167,20 +167,54 @@ func (g *GitGetter) checkout(dst string, ref string) error {
 	return getRunCommand(cmd)
 }
 
+// gitCommitIDRegex is a pattern intended to match strings that seem
+// "likely to be" git commit IDs, rather than named refs. This cannot be
+// an exact decision because it's valid to name a branch or tag after a series
+// of hexadecimal digits too.
+//
+// We require at least 7 digits here because that's the smallest size git
+// itself will typically generate, and so it'll reduce the risk of false
+// positives on short branch names that happen to also be "hex words".
+var gitCommitIDRegex = regexp.MustCompile("^[0-9a-fA-F]{7,40}$")
+
 func (g *GitGetter) clone(ctx context.Context, dst, sshKeyFile string, u *url.URL, ref string, depth int) error {
 	args := []string{"clone"}
 
+	autoBranch := false
 	if ref == "" {
 		ref = findRemoteDefaultBranch(u)
+		autoBranch = true
```

Comment:
i think this is either a very subjective nit or me not understanding (or both), so feel free to ignore:
is `autoBranch` essentially a way to tell if we're using the default branch (so `ref == ""`)? 
if so, i think it might be clearer to either name it a little differently (maybe like `onDefaultBranch` or something) or use `ref == ""` directly.

Reply:
I guess the subtlety I was trying to capture with this name is that we're recording whether the branch was _explicitly chosen_, vs. it being detected automatically. Specifically, the user can potentially write `?ref=main` _even if_ `main` happens to be the default branch name, in which case `autoBranch` would be `false` even though we're going to end up on the default branch in the end.

_Really_ the only motivation for the separate variable here is that we just overwrote the original `ref` on the line above, and so we no longer have the information about whether `ref` was originally `""` or not. I didn't really like it when I wrote it, but was trying to change what was already here as little as possible. With that said, I think it could be clearer for the extra variable to be `originalRef := ref` (a string) and then test if `originalRef == ""` below, which I think more directly captures the intended meaning of "did the caller specify a ref?". What do you think? :thinking:

Reply:
ooooo yeah. i totally missed that we were overwriting ref. 
in that case, i think your proposal of adding `originalRef` captures what you explained really well!